src/main/java/appeng/core/Registration.java

```diff
@@ -150,18 +143,11 @@ void preInitialize( final FMLPreInitializationEvent event )
 
 		MinecraftForge.EVENT_BUS.register( OreDictionaryHandler.INSTANCE );
 
-		final ApiDefinitions definitions = api.definitions();
+		ApiDefinitions definitions = api.definitions();
```

Comment:
Oh wow I didn't actually know about javac doing that optimization, which is interesting. Although for most cases it won't make a difference since we do very little actual arithmetic with fixed values. 
I just think final all over the place hurts readability. Although I practice final fields wherever it is possible, I think they bring little benefit for method parameters and local variables. 

Man you just sent me on a journey to obtain hsdis to look at the actual assembly the JIT generates for your test code :)